src/rules/ban_types.rs

```diff
@@ -1,16 +1,71 @@
 // Copyright 2020-2021 the Deno authors. All rights reserved. MIT license.
-use super::{Context, LintRule, ProgramRef, DUMMY_NODE};
-use once_cell::sync::Lazy;
-use std::collections::HashMap;
-use swc_ecmascript::ast::{
-  TsEntityName, TsKeywordType, TsKeywordTypeKind, TsTypeLit,
-  TsTypeParamInstantiation, TsTypeRef,
+use super::{Context, LintRule, ProgramRef};
+use crate::handler::{Handler, Traverse};
+use dprint_swc_ecma_ast_view::{
+  self as AstView, Spanned, TsEntityName, TsKeywordTypeKind,
 };
-use swc_ecmascript::visit::Node;
-use swc_ecmascript::visit::Visit;
+use if_chain::if_chain;
+use std::convert::TryFrom;
 
 pub struct BanTypes;
 
+const CODE: &str = "ban-types";
+
+#[derive(Clone, Copy)]
+enum BannedType {
+  String,
+  Boolean,
+  Number,
+  Symbol,
+  Function,
+  CapitalObject,
+  LowerObject,
+}
+
+impl BannedType {
+  fn as_message(&self) -> &'static str {
+    use BannedType::*;
+    match *self {
+      String | Boolean | Number | Symbol => "For consistency, it is recommended to use the lower-case primitive",
```

Comment:
Nitpick: Technically it's for more than just consistency.

https://www.typescriptlang.org/docs/handbook/declaration-files/do-s-and-don-ts.html#number-string-boolean-symbol-and-object

Reply:
Thanks for the reference, I didn't read that. Previous wording was based on [the comment in a typescript-eslint example](https://github.com/typescript-eslint/typescript-eslint/blob/79b510c00f25d76daf741936682e1222b0c994b5/packages/eslint-plugin/docs/rules/ban-types.md#examples), but it would be good to refer to the TS handbook :) e727b9c